plugin/trino-hive/src/main/java/io/trino/plugin/hive/avro/AvroPageSourceFactory.java

```diff
@@ -191,11 +193,15 @@ private Schema maskColumnsFromTableSchema(List<HiveColumnHandle> columns, Schema
                 .record(tableSchema.getName())
                 .namespace(tableSchema.getNamespace())
                 .fields();
+        Map<String, String> lowerToGivenName = getLowerCaseToGivenFieldName(tableSchema);
 
         for (String columnName : maskedColumns) {
             Schema.Field field = tableSchema.getField(columnName);
             if (Objects.isNull(field)) {
-                continue;
+                if (!lowerToGivenName.containsKey(columnName)) {
+                    throw new TrinoException(HIVE_CANNOT_OPEN_SPLIT, "Unable to find column %s in table cvro schema %s".formatted(columnName, tableSchema.getFullName()));
```

Comment:
Did you come up with a good explanation why this might happen?

Reply:
I only replicated this by:
1) Fixing the trino writer path first
2) Testing the reader path and getting errors.